org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceExecuteCommandHandler.java

```diff
@@ -219,4 +226,32 @@ public void handleException(Throwable ex) {
 		});
 		return resultValues[0];
 	}
+
+	@Override
+	public synchronized void added(IExtension[] extensions) {
+		removed(extensions);

```

Comment:
Remove the extensions that are already registered so they aren't duplicated. Since I don't know if removed notifications is always sent when you install a new version of the bundle  (i'm new to OSGi), it is a cheap safeguard.

It is safe to remove this line?

Reply:
yeah, it's ok to remove this line.